### PR TITLE
feat: catch failures + support laravel 6 + improvements

### LINKED ISSUE #1: Can't install on Laravel 6
Hello, the package seems to be cool unfortunately I cannot install it on Laravel 6

### LINKED ISSUE #4: Detect composer installation failure
Using this on another project :D
Using my PR for l6, #2 but I think this applies anyway.

If composer install fails, the installer continues, but the app is left in a broken state. 

We should instead catch the failure exit code, and present the error to fix to the user.

Happy to PR :)

### PR DESCRIPTION
**Pull request by atymic**
Replaces my previous PR, #2 

Note: This needs https://github.com/Harmonic/inertia-table/pull/17 to be merged first

I've made a bunch of changes, namely: 

- Added a quick script to spin up a new install for testing
- Switched execution of commands to `Symfony\Process` package, meaning that exceptions will be thrown on failure, so the errors are helpful (see screenshot)
- Fixed incorrectly named service provider
- Support laravel 6
- Fix formatting/typos + type hints

Closes #4
Closes #1

### CODE REVIEW

File: src/Preset.php [hunk 1]
```diff
@@ -156,7 +157,11 @@ public function run(): void
                 copy(__DIR__ . '/stubs/.eslintrc.js', base_path('.eslintrc.js'));
             });
             $this->command->task('Setup tailwindcss', function () {
-                $this->runCommand('yarn tailwind init');
+                try {
+                    $this->runCommand('yarn tailwind init');
+                } catch (ProcessFailedException $e) {
+                    // Tailwind already setup by the theme
```
Comment (craigharman) [inline src/Preset.php]: Suggest we notify the user here that Tailwind already detected.
Are their other circumstances ProcessFailedException would be triggered?

Comment (atymic) [reply src/Preset.php]: I figured it probably wasn't worth writing a line for (as it only happens if you say yes to inertia, which stubs out the tailwind file.

Will update now.

**Comment by atymic** [comment]
> Are their other circumstances ProcessFailedException would be triggered?

I ran it through a few times without issue. The `ProcessFailedException` will be thrown any time a command fails, but I think this is the correct behaviour (for example if you don't configure your DB, it would fail on the migrations step)

Ideally (maybe a future PR) it would remember the step it failed at and continue from there (since you can't run some steps twice due to the stubs)

**Comment by atymic** [comment]
@craigharman this currently isn't working because the PR for intertia table hasn't been merged:
https://github.com/Harmonic/inertia-table/pull/17

**Pull request merged**